back/src/domain/convention/useCases/UpdateConventionStatus.ts

```diff
@@ -60,50 +60,42 @@ export class UpdateConventionStatus extends TransactionalUseCase<
     uow: UnitOfWork,
     { conventionId, role }: UpdateConventionStatusPayload,
   ): Promise<WithConventionId> {
-    const { status } = params;
-    logger.debug({ status, conventionId, role });
+    logger.debug({ status: params.status, conventionId, role });
 
     const conventionUpdatedAt = this.timeGateway.now().toISOString();
 
+    const statusJustification =
+      params.status === "CANCELLED" ||
+      params.status === "REJECTED" ||
+      params.status === "DRAFT" ||
+      params.status === "DEPRECATED"
+        ? params.statusJustification
+        : undefined;
+
```

Comment:
tu as des utilitaires qui peuvent t'aider dans `convention.dto.ts`

Genre `doesStatusNeedsJustification`, ou tout simplement :
```suggestion
    const statusJustification = conventionStatusesWithJustification.includes(params.status)
        ? params.statusJustification
        : undefined;
```

Reply:
J'avais essayé mais ça ne passe pas niveau TS. Surement un problème de scope sur le status qui ne s'applique pas sur le reste de la convention.

![image](https://github.com/gip-inclusion/immersion-facile/assets/58432308/aec74a6b-889f-4ef6-9963-4350980e5466)

Reply:
<img width="491" alt="image" src="https://github.com/gip-inclusion/immersion-facile/assets/22095555/1308f33c-e642-4918-b6fc-e8d801f5f04a">